src/components/cylc/Graph.vue

```diff
@@ -0,0 +1,1329 @@
+  },
+
+  methods: {
+    subscribe (queryName) {
+      const id = workflowService.subscribe(
+        this,
+        QUERIES[queryName].replace('WORKFLOW_ID', this.workflowId),
+        this.setActive
+      )
+      if (!(queryName in this.subscriptions)) {
+        this.subscriptions[queryName] = {
+          id,
+          active: false
+        }
+      }
+    },
+
+    unsubscribe (queryName) {
+      if (queryName in this.subscriptions) {
+        workflowService.unsubscribe(
+          this.subscriptions[queryName].id
+        )
+      }
+    },
+
+    setActive (isActive) {
+      this.isLoading = !isActive
+    },
+
+    async workflowUpdated (workflows) {
+      try {
+        console.log('workflowUpdated')
+        const elements = {
+          nodes: [],
+          edges: []
+        }
+        if (!isUndefined(workflows)) {
+          each(workflows, (value, key) => {
+            each(value, (workflow, key) => {
+              if (has(workflow.nodesEdges, 'edges') && !isUndefined(workflow.nodesEdges.edges)) {
+                elements.edges = this.getEdges(workflow.nodesEdges.edges)
+              }
+              if (has(workflow.nodesEdges, 'nodes') && !isUndefined(workflow.nodesEdges.nodes)) {
+                elements.nodes = this.getNodes(workflow.nodesEdges.nodes)
+              }
+            })
+          })
+        }
+        // console.log('elements ==>>>> ', elements)
+        isEmpty(elements) ? console.log('gdata is empty or undefined') : this.graphData = elements
+      } catch (error) {
+        console.error('workflowUpdated error: ', error)
+      }
+    },
+
+    getNodes (nodes) {
+      try {
+        const nodesArray = []
+        let nodeObj = {}
+        let parentId = ''
+        each(nodes, (node, key) => {
+          nodeObj = {
+            data: {
+              id: '',
+              runpercent: 0,
+              state: 'undefined',
+              parent: '',
+              label: '',
+              shape: 'ellipse'
+            },
+            position: {
+
+            },
+            group: 'nodes',
+            removed: false,
+            selected: false,
+            selectable: true,
+            locked: false,
+            grabbable: true,
+            classes: ''
+          }
+          has(node, 'id') && !isEmpty(node.id) ? nodeObj.data.id = getUuid(node.id) : console.log('workflowUpdated - node id is empty')
+          has(node, 'label') && !isEmpty(node.label) ? nodeObj.data.label = node.label : console.log('workflowUpdated - node label is empty')
+          has(node, 'state') && !isEmpty(node.state) ? nodeObj.data.state = node.state : nodeObj.state = 'undefined'
+          has(node, 'runpercent') && !isEmpty(node.runpercent) ? nodeObj.data.runpercent = node.runpercent : nodeObj.data.runpercent = 0
+          has(node, 'parent') && has(node.parent, 'id') && !isEmpty(node.parent.id) ? parentId = getUuid(node.parent.id)
+            : console.log('workflowUpdated - node parent is empty')
```

Comment:
Just a bit of a style here. Given that you are not assigning results of the ternary expressions, I would imagine that `if ... else` blocks would be easier to read here.

Reply:
I suppose it is subjective. I prefer ternary when it can be a simple one liner, the results are assigned to properties in the node object.

Reply:
Don't get me wrong, I am a big fan of the ternary operator myself. In my life before Python, I have used it a lot to write simple logic tables, but I would always have the result assigned to a variable.

Yes, it is perhaps just a style preference.

Reply:
We are using `console.debug` in other parts of the code I think. Perhaps we could use that here too? My preference is also to have both sides of the ternary operator to be similar... i.e. `condition ? log(a) : log(b)` or `var x = condition ? a : b`. But not a blocker IMO if @MartinRyan prefers to keep it :+1: